### PR TITLE
python2Packages.supervisor: 3.1.4 -> 3.3.4

### PR DESCRIPTION
**Pull request by zimbatm**
###### Motivation for this change

Package upgrade.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by zimbatm** [comment]
I guess this is going to have the same problem of python env var leaks to sub-processes right @FRidh ?

File: pkgs/development/python-modules/supervisor/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+buildPythonPackage rec {
+  pname = "supervisor";
+  version = "3.3.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0wp62z9xprvz2krg02xnbwcnq6pxfq3byd8cxx8c2d8xznih28i1";
+  };
+
+  buildInputs = [ mock ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/supervisor/default.nix]: `checkInputs`

File: pkgs/development/python-modules/supervisor/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  propagatedBuildInputs = [ meld3 ];
+
+  # Supervisor requires Python 2.4 or later but does not work on any version of Python 3.  You are using version 3.6.5 (default, Mar 28 2018, 10:24:30)
+  disabled = isPy3k;
+
+  meta = {
+    description = "A system for controlling process state under UNIX";
+    homepage = http://supervisord.org/;
+    maintainers = with lib.maintainers; [ zimbatm ];
+  };
```
Comment (dotlambda) [inline pkgs/development/python-modules/supervisor/default.nix]:
```
license = lib.licenses.free; # http://www.repoze.org/LICENSE.txt
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by zimbatm** [comment]
@dotlambda thanks, fixed. learned about checkInputs today.

@GrahamcOfBorg build python2Packages.supervisor

**Review by dotlambda — APPROVED** [review]

**Comment by FRidh** [comment]
@zimbatm what are you specifically referring to?

**Comment by zimbatm** [comment]
@FRidh nevermind, the python wrapper doesn't set the python lib path through an environment variable anymore

**Pull request merged**